src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1311 @@
+        Return an iterator for the mutation class of this seed.
+
+        INPUT:
+
+        - ``depth`` -- Integer (default: ``infinity``); only return seeds at
+          most ``depth`` mutations away from the initial seed
+
+        - ``verbose`` -- Boolean (default: ``False``); if ``True``, the
+          current depth of recursion for the chosen algorithm is shown while
+          computing
+
+        - ``return_paths`` -- Boolean (default: ``False``); if ``True``, a
+          path of mutations from ``self`` to the given seed is returned as well
+
+        - ``algorithm`` -- String (default: ``'BFS'``); the search algorithm to
+          find new seeds. Currently supported options::
+
+          * 'BFS' - breadth-first search
+          * 'DFS' - depth-first search
+
+        EXAMPLES:
+
+        We iterate over the mutation class for a rank two seed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: t = S.mutation_class_iter()
+            sage: for seed in t: print(seed.cluster())
+            [x1, x2]
+            [(x2 + 1)/x1, x2]
+            [x1, (x1 + 1)/x2]
+            [(x2 + 1)/x1, (x1 + x2 + 1)/(x1*x2)]
+            [(x1 + x2 + 1)/(x1*x2), (x1 + 1)/x2]
+
+        Non finite-type works if we specify a fixed depth, but seeds can get big
+        rather quickly::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2 + x2^2, x2: 1 + x1 + x1^2})
+            sage: t = S.mutation_class_iter(depth=15,algorithm='DFS')
+            sage: for seed in t: print(seed.cluster()[0].denominator()) # long time
+            1
+            x1
+            1
+            x1*x2^2
+            x1*x2^2
+            x1^3*x2^4
+            x1^3*x2^4
+            x1^5*x2^6
+            x1^5*x2^6
+            x1^7*x2^8
+            x1^7*x2^8
+            x1^9*x2^10
+            x1^9*x2^10
+            x1^11*x2^12
+            x1^11*x2^12
+            x1^13*x2^14
+
+        We can print computational statistics when computing examples::
+
+            sage: var('x1,x2,x3,a0,a2,a3,b0,b1,b3,c0,c1,c2')
+            (x1, x2, x3, a0, a2, a3, b0, b1, b3, c0, c1, c2)
+            sage: S = LPASeed({x1: a0 + a2*x2 + a3*x3, x2: b0 + b1*x1 + b3*x3, x3: c0 + c1*x1 + c2*x2})
+            sage: t = S.mutation_class_iter(verbose=True)
+            sage: for seed in t: None
+            depth: 0     found: 1
+            depth: 1     found: 4
+            depth: 2     found: 10
+            depth: 3     found: 16
+        """
+
+        if algorithm == 'BFS':
+
+            return self._mutation_class_iter_bfs(
+                depth=depth, verbose=verbose, return_paths=return_paths)
+
+        elif algorithm == 'DFS':
+
+            return self._mutation_class_iter_dfs(
+                depth=depth, verbose=verbose, return_paths=return_paths)
+
+        else:
+
+            raise ValueError('nonsupported search algorithm: %s' % (algorithm))
```

Comment:
```suggestion
        """
        if algorithm == 'BFS':
            return self._mutation_class_iter_bfs(
                depth=depth, verbose=verbose, return_paths=return_paths)

        elif algorithm == 'DFS':
            return self._mutation_class_iter_dfs(
                depth=depth, verbose=verbose, return_paths=return_paths)

        raise ValueError('nonsupported search algorithm: %s' % (algorithm))
```
IMO this is more clearly indicates that everything else falls into the `ValueError`.